### DISCUSSION TITLE
Is there any way I can get the prompt as a string?

### DISCUSSION THREAD

**Discussion by UtkarshVerma** [romkatv/powerlevel10k#2606] [General]
I am writing a script for which I need to know the prompt string (`~/music >`) value as a string. Is there any way I can print it in zsh and store it in a variable?

This part:
![image](https://github.com/romkatv/powerlevel10k/assets/31820255/600494ea-1dfd-4946-8e0b-a9b65cd6b4b9)

**Comment by romkatv** [comment]
Powerlevel10k does not provide an API for this. Depending on what exactly you are after, you might be able to achieve it through lower level APIs.

**Reply by UtkarshVerma** [reply]
I am writing a script to filter the commands from the entire terminal output. I plan to do it by `grep`ping all lines starting with the prompt string. Do you think there's any other way to do this?

**Comment by romkatv** [comment]
If you are capturing the complete TTY output, you can enable terminal integration in powerlevel10k (`POWERLEVEL9K_TERM_SHELL_INTEGRATION=true`) and identify prompt by its markers: `\e]133;A\a` and `\e]133;B\a`.

If you are capturing only text, the easiest solution is to identify prompt by a special symbol that is present in all prompt but unlikely to be present elsewhere. Given your screenshot, I assume that all your prompts have `❯` or `❮` in them. You can grep that.

**Reply by UtkarshVerma** [reply]
Thank you, this was exactly what I wanted!
